lib/Importer/ONNXModelLoader.cpp

```diff
@@ -1075,12 +1078,28 @@ ONNXModelLoader::loadConstantOfShape(const ONNX_NAMESPACE::NodeProto &op,
     auto TH = in->getPayload().getHandle<int64_t>();
     llvm::ArrayRef<size_t> outputDims = {(const size_t *)TH.begin(),
                                          (const size_t *)TH.end()};
-    ty = G_.getParent()->uniqueType(ElemKind::FloatTy, outputDims);
+    switch (T.getType().getElementType()) {
+    case ElemKind::FloatTy:
+      ty = G_.getParent()->uniqueType(ElemKind::FloatTy, outputDims);
+      SN = G_.createSplat(loadOperatorName(op), ty, T.getHandle().raw(0));
+      break;
+    case ElemKind::Int64ITy:
+      ty = G_.getParent()->uniqueType(ElemKind::Int64ITy, outputDims);
```

Comment:
can you lift this up before switch (you have elem kind from T.getType().getElementType()).

Reply:
I didn't lift it up to take care of the the default scenario, which now kind of looks redundant because we already have a check in the beginning of the function. Will do that and remove the default scenario.